tests/scheduler/test_detailers.py

```diff
@@ -425,6 +425,65 @@ def test_region_label(self):
             assert "lowdust" in res["target_name"]
             assert "nes" in res["target_name"]
 
+    def test_euclid_dither(self):
+        detailer = detailers.EuclidDitherDetailer(per_night=True)
+
+        ra_a, dec_a, ra_b, dec_b = detailer._generate_offsets(100, 2)
+
+        # Going per night, should only be one offset
+        assert np.size(ra_a) == 1
+
+        # if we call again, should be the same
+        ra_ap, dec_ap, ra_bp, dec_bp = detailer._generate_offsets(100, 2)
+
+        assert ra_a == ra_ap
+        assert dec_a == dec_ap
+        assert ra_b == ra_bp
+        assert dec_b == dec_bp
+
+        # Now want a unique position
+        detailer = detailers.EuclidDitherDetailer(per_night=False)
+
+        ra_a, dec_a, ra_b, dec_b = detailer._generate_offsets(100, 2)
+
+        assert np.size(ra_a) == 100
+
+        # call again should result in different values
+        ra_ap, dec_ap, ra_bp, dec_bp = detailer._generate_offsets(100, 2)
+        assert np.all(np.not_equal(ra_a, ra_ap))
+
+    def test_dither(self):
+        detailer = detailers.DitherDetailer(per_night=True)
+
+        offsets = detailer._generate_offsets(100, 2)
+        ra_a = offsets[:, 0]
+        dec_a = offsets[:, 1]
+        # Going per night, should only be one offset
+        assert np.size(np.unique(ra_a)) == 1
+
+        # if we call again, should be the same
+        offsets = detailer._generate_offsets(100, 2)
+        ra_ap = offsets[:, 0]
+        dec_ap = offsets[:, 1]
+        assert np.all(ra_a == ra_ap)
+        assert np.all(dec_a == dec_ap)
+
+        # Now want a unique position
+        detailer = detailers.DitherDetailer(per_night=False)
+
+        offsets = detailer._generate_offsets(100, 2)
+        ra_a = offsets[:, 0]
+        dec_a = offsets[:, 1]
+
+        assert np.size(np.unique(ra_a)) == 100
+
+        # call again should result in different values
```

Comment:
I'm not sure I understand this -- if the dither detailer is called per_night=True, it pre-generates all of the offsets, so the values are fixed? But if you call it per_night=False, then it does not pre-generate all of the offsets, so they change? 
This is probably not unreasonable (I imagine you wanted to avoid saving all of the dithers for the entire survey, if they were per-visit). It should be documented however (there's no mention of this in the doc-string). 

The other thing I was wondering about was if the FBS was re-enabled during the night and doing per-visit dithers, this means the dithers would repeat within the night? Again also probably not terrible, but likely something to make sure is documented as expected behavior.

Reply:
It is always pre-generating positions. This is testing that `per_night` behaves properly and gives only one offset value if `True` and a unique position per visit if `False`.

Depends on what "re-enabled" means. If you re-instantiate the scheduler and play in completed observations, then we lose the count and potentially repeat dither positions. If you restore from a pickle, then the count is fine and there shouldn't be repeats.

Reply:
We talked about this on slack, but there were two issues: 
I didn't quite understand the API  for `_generate_offsets` and thought calling it with the same numbers meant the same values (but instead, internally this method increments a counter). 
The second issue is that an internal counter in a detailer works for simulations but probably doesn't work at the summit, since anything that happens as a result of `request_observations` is lost in the pickle/restore cycle. Only the results of `update_conditions` and `add_observation` are kept and updated in the FBS.

Reply:
Adding a `feature` has resolved the problem I think.